File: .agentic/templates/features/auth/impl/java-quarkus/server/src/main/java/com/example/app/auth/security/Passwords.java
package com.example.app.auth.security;

import io.quarkus.elytron.security.common.BcryptUtil;

/** bcrypt password hashing (constitution F2 — no custom crypto). Shared by both strategies. */
public final class Passwords {

    private Passwords() {
    }

    public static String hash(String plain) {
        return BcryptUtil.bcryptHash(plain, 12);
    }

    public static boolean verify(String plain, String hash) {
        return BcryptUtil.matches(plain, hash);
    }
}

File: .agentic/templates/features/auth/impl/python-fastapi/server/src/auth/passwords.py
from passlib.context import CryptContext

# bcrypt (constitution F2 — no custom crypto). Shared by both token strategies.
_pwd = CryptContext(schemes=["bcrypt"], deprecated="auto", bcrypt__rounds=12)


def hash_password(plain: str) -> str:
    return _pwd.hash(plain)


def verify_password(plain: str, hashed: str) -> bool:
    return _pwd.verify(plain, hashed)
